gramhopper/configuration/rules_parser.py

```diff
@@ -30,10 +34,38 @@ def parse_globals(self, config: CommentedMap):
         if 'responses' in config:
             self.global_responses.update(ResponseParser.parse_many(config['responses']))
 
-    def _parse_rule_trigger_or_response(self, rule: CommentedMap, params: TriggerOrResponseParams):
-        # If the trigger/response here is just a name, look for it in the globals
+    def _parse_and_or_expression(self,
```

Comment:
You wrote a lot of code to parse the expression, but if all we're expecting is a sequence of boolean operators and trigger names, I believe you can just replace any operator to lower case (to transform it into python operator), and pull the trigger from the globals dict, and just `eval()` the entire expression as is.

Reply:
Oh I really don't wanna use python's `eval()`. But as we've discussed this morning, I think the boolean algebra package has an evaluation function that will solve that.